Mathlib/Combinatorics/Additive/SubsetSum.lean

```diff
@@ -0,0 +1,119 @@
+/-
+Copyright (c) 2025 Aviv Bar Natan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Aviv Bar Natan
+-/
+import Mathlib.Algebra.BigOperators.Group.Finset.Basic
+import Mathlib.Algebra.Order.Monoid.Canonical.Defs
+import Mathlib.Data.Finset.Max
+import Mathlib.Data.Finset.Powerset
+
+/-!
+# Subset sums
+
+This file defines the subset sum of a finite subset of a commutative monoid.
+
+## References
+
+* [Melvyn B. Nathanson, *Inverse theorems for subset sums*][Nathanson1995]
+-/
+
+namespace Finset
+variable {α : Type*} [DecidableEq α] [AddCommMonoid α] {A : Finset α}
```

Comment:
Please avoid using generic variables (greek letters) for types equipped with algebraic structure
```suggestion
variable {M : Type*} [DecidableEq M] [AddCommMonoid M] {A : Finset M} {a : M}
```